src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,214 @@
+$outputFilePath=".\Detail_Account_196312_202309_en_filtered.csv"
+
+# Define the column mapping
+# The keys are the original column names, and the values are the new column names. If the value is the same as the key, the column will not be renamed.
+# If the value is not the same as the key, the column will be renamed and the value will be used as the new column name.
+# If the key is not present in the input data, the column will not be included in the output data.
+$columnMapping = @{
+    billingAccountId = 'BillingAccountId'
+    billingAccountName = 'BillingAccountName'
+    ftk_AccountType = 'ftk_AccountType'
+    BillingProfileId = 'BillingProfileId'
+    SubscriptionId = 'SubscriptionId'
+}
+
+<# Helper Functions #>
+function Add-AccountType {
+    [CmdletBinding()]
+    param 
+    (
+        [PSCustomObject]
+        $Object
+    )
+
+    $ftk_AccountType = switch ($true) {
+        { $Object.BillingAccountId -eq $Object.BillingProfileId } { "EA" }
+        { $null -eq $Object.BillingProfileId -or 
+          $Object.BillingProfileId -eq $Object.SubscriptionId } { "Other" }
+        default { "MCA" }
+    }
+    # Add or Update the ftk_AccountType property to the object.
+    if ($Object.PSObject.Properties["ftk_AccountType"]) {
+        $Object.PSObject.Properties["ftk_AccountType"].Value = $ftk_AccountType
+    } else {
+        $Object | Add-Member -MemberType NoteProperty -Name "ftk_AccountType" -Value $ftk_AccountType -Force
+    }
+    
+    return $Object
+}
+
+
+<# Main Function #>
+function ConvertTo-FinOpsSchema {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter()]
+        [string]
+        $ActualCost,
+
+        [Parameter()]
+        [string]
+        $outputFilePath,
+
+        [Parameter()]
+        [hashtable]
+        $columnMapping,
+
+        [Parameter()]
+        [bool]
+        $ExportAllColumns,
+
+        [Parameter()]
+        [switch]
+        $Force
+    )
+    
+    if (-not (Test-Path -Path $ActualCost)) {
+        Write-Error "Input file $ActualCost does not exist."
+        return
+    }
+    
+    $outputDirectory = [System.IO.Path]::GetDirectoryName($outputFilePath)
+    if (-not (Test-Path -Path $outputDirectory -PathType Container)) {
+        New-Item -Path $outputDirectory -ItemType Directory
+    }
+
+    # Import the CSV data
+    $csvData = Import-Csv -Path $ActualCost
+    
+    # Iterate over each row and apply transformations
+    $transformedData = $csvData | ForEach-Object {
+        $object = $_
+        $object = Add-AccountType -Object $object
+
+        # Create a new object with the mapped column names
+        $newObject = @{}
+        foreach ($column in $object.PSObject.Properties.Name) {
+            if ($ExportAllColumns -or $column -in $columnMapping.Keys) {
+                $newColumnName = if ($columnMapping.ContainsKey($column)) { $columnMapping[$column] } else { $column }
+                $newObject[$newColumnName] = $object.$column
+            }
+        }
```

Comment:
Would it be simpler to replace the mapping config with properties in the new object declaration? This should be faster, too.
```powershell
        # Create a new object with the mapped column names
        $newObject = @{
            BillingAccountId = $object.BillingAccountId
            BillingAccountName = $object.BillingAccountName
            SubAccountId = $object.SubscriptionId
            SubAccountName = $object.SubscriptionName

            # If we have an option to exclude the custom columns, we'll need to move this out of the definition
            ftk_AccountType = (Get-AccountType $object)
            ftk_BillingProfileId = $object.BillingProfileId
            ...
        }
```
_NOTE: This will get more advanced as we add other schemas, but this is the basic version._

The only thing I don't love about this is it doesn't account for columns we don't know about. So if CM adds a new column, we'd lose it. I'd probably just add a `# TODO` for that and we can figure it out later.

Reply:
Done the change. Please do test from your side to make sure this is what I understand from the requirement and that it is correct.